src/chromatix/field.py

```diff
@@ -225,6 +235,55 @@ def conj(self) -> Array:
         """conjugate of the complex field, as a field of the same shape."""
         return self.replace(u=jnp.conj(self.u))
 
+    @property
+    def spatial_limits(self) -> Tuple[Tuple[float, float], Tuple[float, float]]:
+        """
+        Return the spatial limits of the field: (y_min, y_max), (x_min, x_max).
+        """
+        return (float(self.grid[0].min()), float(self.grid[0].max())), (
+            float(self.grid[1].min()),
+            float(self.grid[1].max()),
+        )
+
+    def shift(self, shift_yx: Union[float, Array]) -> Field:
+        """
+        Shift the field in the destination plane.
+
+        Args:
+            shift_yx: The shift in the destination plane. Should be an array of
+                shape `[2,]` in the format `[y, x]`.
+        """
+        if isinstance(shift_yx, Number):
+            shift_yx = jnp.array([shift_yx, shift_yx])
+        shift_yx = jnp.array(shift_yx)  # Ensure it is an array
+        if shift_yx.ndim == 1:
+            shift_yx = shift_yx[:, None]
+        assert_rank(shift_yx, 2)
+        return self.replace(_shift_yx=shift_yx)
+
+    def set_sampling(
```

Comment:
@diptodip I added this method to set the sampling of a field, which will overwrite `u` with an empty field, as the new sampling size may not be compatible with the new shape

Reply:
Same thing re: functions as the previous comment. But this name also sounds like we might do resampling of the `Field` here, which might be confusing. So I think then that this naming is unclear, maybe it should be something like a class method that says `ScalarField.empty_like(some_other_field)`/`VectorField.empty_like(some_other_field)` and work like the functions that create new `Field`s.

Reply:
done 👍